repoman/git/depot_operations.py

```diff
@@ -50,24 +50,31 @@ def grab_changesets(self, path, url, changesets):
         logger.debug('Grabbing changesets from %s to %s' % (url, path))
         git_repo = pygit2.Repository(path)
         try:
-            origin = self._set_origin_source(git_repo, url)
-            #this may fail against github
-            result = origin.fetch()
-            logger.debug('GIT Done grabbing changesets (%s)' % (result))
-        except (pygit2.GitError, OSError) as e:
-            logger.exception('Error Grabbing changesets: %s' % e)
-            logger.info("Retrying using process")
-            try:
+            self._set_origin_source(git_repo, url)
```

Comment:
After setting origin, cannot we just `git reset --hard` if it is not bare and then `git fetch origin` in any case?